packages/hop-node/src/watchers/SyncWatcher.ts

```diff
@@ -223,138 +256,199 @@ class SyncWatcher extends BaseWatcher {
+    if (
+      !this.isInitialSyncCompleted() ||
+      this.syncIndex % SyncCyclesPerFullSync === 0
+    ) {
+      promisesPerPoll = this.getAllPromises()
+    } else {
+      promisesPerPoll = this.getTransferSentPromises()
+    }
 
+    // these must come after db is done syncing, and syncAvailableCredit must be last
+    await Promise.all(promisesPerPoll)
+      .then(async () => await this.availableLiquidityWatcher.syncBonderCredit())
+  }
+
+  getSyncOptions (keyName: string) {
     // Use a custom start block number on the initial sync if it is defined
     let startBlockNumber: number = this.bridge.bridgeDeployedBlockNumber
     if (!this.isInitialSyncCompleted() && this.customStartBlockNumber) {
       startBlockNumber = this.customStartBlockNumber
     }
 
-    const getOptions = (keyName: string) => {
-      return {
-        syncCacheKey: this.syncCacheKey(keyName),
-        startBlockNumber
-      }
+    return {
+      syncCacheKey: this.syncCacheKey(keyName),
+      startBlockNumber
     }
+  }
 
-    const transferRootInitialEventPromises: Array<Promise<any>> = []
-    if (this.isL1) {
-      const l1Bridge = this.bridge as L1Bridge
-      transferRootInitialEventPromises.push(
-        l1Bridge.mapTransferRootBondedEvents(
-          async (event: TransferRootBondedEvent) => {
-            return await this.handleTransferRootBondedEvent(event)
-          },
-          getOptions(l1Bridge.TransferRootBonded)
-        )
-      )
+  getTransferSentToL2EventPromise (): Promise<any> {
+    if (!this.isL1) {
+      return Promise.resolve()
+    }
+    const l1Bridge = this.bridge as L1Bridge
+    return l1Bridge.mapTransferSentToL2Events(
+      async (event: TransferSentToL2Event) => {
+        return await this.handleTransferSentToL2Event(event)
+      },
+      this.getSyncOptions(l1Bridge.TransferSentToL2)
+    )
+  }
 
-      promises.push(
-        l1Bridge.mapTransferSentToL2Events(
-          async (event: TransferSentToL2Event) => {
-            return await this.handleTransferSentToL2Event(event)
-          },
-          getOptions(l1Bridge.TransferSentToL2)
-        )
-      )
+  getTransferRootConfirmedEventPromise (): Promise<any> {
+    if (!this.isL1) {
+      return Promise.resolve()
+    }
+    const l1Bridge = this.bridge as L1Bridge
+    return l1Bridge.mapTransferRootConfirmedEvents(
+      async (event: TransferRootConfirmedEvent) => {
+        return await this.handleTransferRootConfirmedEvent(event)
+      },
+      this.getSyncOptions(l1Bridge.TransferRootConfirmed)
+    )
+  }
 
-      promises.push(
-        l1Bridge.mapTransferRootConfirmedEvents(
-          async (event: TransferRootConfirmedEvent) => {
-            return await this.handleTransferRootConfirmedEvent(event)
-          },
-          getOptions(l1Bridge.TransferRootConfirmed)
-        )
-      )
+  getTransferBondChallengedEventPromise (): Promise<any> {
+    if (!this.isL1) {
+      return Promise.resolve()
```

Comment:
suggestion: add `async` to this function and return empty array instead of `Promise.resolve()`


```suggestion
      return []
```